### DISCUSSION TITLE
Migration guide

### DISCUSSION THREAD

**Discussion by noraj** [ppoffice/hexo-theme-icarus#1240] [Q&A]
I was using `hueman` theme that was recently archived (as other themes), I guess to focus the work on icarus theme. So I would be interested about advices on how to migrate from hueman to icarus while keeping the maximum from the old theme (configuration, customization) and know if tehre are some major stuff that existed in hueman that don't anymore in icarus.

**Comment by JIeJaitt** [comment]
In the "source" directory of your Hexo source code files, you have crucial data for each article you've written. Other elements might seem less significant, as styles and similar aspects can be replaced with a new theme. Additionally, refrain from deleting your old site. Once you've successfully migrated to the Icarus blog, consider whether to remove your old site. Before initiating the migration, ensure you have a backup in place.

**Comment by noraj** [comment]
Features still existing in Icarus or having an equivalent:

- categories and tags archive
- Insight search
- social media bar
- Disqus comments
- widgets (Archives Categories External Site Links Recent Posts Tags Table of Contents)
- Update Time

Features that exist in Hueman and seem missing in Icarus:

- sub-menus in the navigation bar ([categories](https://github.com/ppoffice/hexo-theme-hueman#categories-inside-main-menu) or custom links, cf https://github.com/ppoffice/hexo-theme-hueman/pull/288)
  - [ ] https://github.com/ppoffice/hexo-theme-icarus/issues/1250
- [Multi author](https://github.com/ppoffice/hexo-theme-hueman/wiki/Additional-Features) (cf. https://github.com/ppoffice/hexo-theme-hueman/pull/233)
  - [ ] https://github.com/ppoffice/hexo-theme-icarus/issues/1251
- two columns index gallery
  - [ ] https://github.com/ppoffice/hexo-theme-icarus/issues/1252
- container-inner-max-width configurable (cf. https://github.com/ppoffice/hexo-theme-hueman/pull/280)
  - [ ] https://github.com/ppoffice/hexo-theme-icarus/issues/1254
- bugs
  - [ ] https://github.com/ppoffice/hexo-theme-icarus/issues/1249

I guess the posts permalink will stay the same since it's managed by hexo and not the theme?

**Reply by JIeJaitt** [reply]
Some personal opinions.
1. Features not supported by Icarus may require manual exploration and implementation on your part.
2. The permanent link of an article is controlled by the formatter within the article and is independent of the Icarus theme. This functionality is achieved through a Hexo plugin and is associated with the Hexo framework.
